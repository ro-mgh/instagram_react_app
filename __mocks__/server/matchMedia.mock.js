Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
// window.matchMedia = jest.fn().mockImplementation((query) => {
//   return {
//     matches: true,
//     media: query,
//     onchange: null,
//     addListener: jest.fn(), // deprecated
//     removeListener: jest.fn(), // deprecated
//     addEventListener: jest.fn(),
//     removeEventListener: jest.fn(),
//     dispatchEvent: jest.fn(),
//   };
// });
