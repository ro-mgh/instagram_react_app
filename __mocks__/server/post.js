const post = {
  author: {
    avatar:
      "https://insta-project.s3-ap-northeast-2.amazonaws.com/waladPiWs1XukX995ZBlSNODoTH3.jpeg",
    id: "waladPiWs1XukX995ZBlSNODoTH3",
    username: "burningman",
  },
  comments: [],
  createdAt: "2021-02-19T09:04:02.437Z",
  id: 49,
  image:
    "https://insta-project.s3-ap-northeast-2.amazonaws.com/waladPiWs1XukX995ZBlSNODoTH3&&1613725435570.jpeg",
  likes: [],
  text: null,
};

export const _post = post;
