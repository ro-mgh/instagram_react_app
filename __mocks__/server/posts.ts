const posts = [
  {
    createdAt: "2021-02-07T10:35:12.443Z",
    id: 27,
    image:
      "https://insta-project.s3-ap-northeast-2.amazonaws.com/hPpgfQK9i3MC1ATiW1BHP7jIDai1&&1612694084227.jpeg",
  },
  {
    createdAt: "2021-02-04T16:33:31.034Z",
    id: 26,
    image:
      "https://insta-project.s3-ap-northeast-2.amazonaws.com/hPpgfQK9i3MC1ATiW1BHP7jIDai1&&1612456403166.jpeg",
  },
  {
    createdAt: "2021-02-04T16:33:18.032Z",
    id: 25,
    image:
      "https://insta-project.s3-ap-northeast-2.amazonaws.com/hPpgfQK9i3MC1ATiW1BHP7jIDai1&&1612456392129.jpeg",
  },
];

export const _posts = posts;
