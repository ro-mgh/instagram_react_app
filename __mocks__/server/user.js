export const _user = {
  id: "123",
  avatar: "",
  username: "test",
  name: "testName",
};
