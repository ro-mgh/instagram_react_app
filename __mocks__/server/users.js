export const _users = {
  SOMEID: {
    id: "id",
    username: "test",
    name: "testName",
    avatar: "",
    followingIds: [],
    Following: [],
    posts: [],
  },
};
