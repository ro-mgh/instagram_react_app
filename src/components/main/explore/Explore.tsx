import React, { useEffect } from "react";
import FooterBottom from "../../footer/FooterBottom";
import Carousel from "./Carousel";
import Header from "../header/Header";
import Gallery from "./Gallery";
// import { explorePosts } from "../../../store/actions/exploreUsers";
// import { useDispatch, useSelector } from "react-redux";
import CircularProgress from "@material-ui/core/CircularProgress";
import { useQuery } from "react-query";
import { fetchNotFollowingPosts } from "../../../utils/fetchingData";

const Explore = () => {
  // const usersPostsArr = useSelector((state) => state.dataReducer.usersPosts);
  // const dispatch = useDispatch();

  const { isLoading, error, data } = useQuery(
    "notFollowingPosts",
    fetchNotFollowingPosts
  );

  if (error) {
    console.log("An error has occurred: " + error.message);
  }

  // useEffect(() => {
  //   dispatch(explorePosts());
  // }, []);

  return (
    <div>
      <Header />
      <div className="explore-wrapper">
        <div className="explore-users-wrapper">
          <p className="explore-users-text">Discover People</p>
          <div className="explore-carousel-wrapper">
            <Carousel />
          </div>
        </div>
        <div className="explore-posts-wrapper">
          <p className="explore-users-text">Explore</p>
          {!isLoading ? (
            <Gallery {...{ posts: data }} />
          ) : (
            <div className="mainfield-progress-wrapper">
              <CircularProgress size={30} />
            </div>
          )}
        </div>
      </div>
      <FooterBottom />
    </div>
  );
};

export default Explore;
