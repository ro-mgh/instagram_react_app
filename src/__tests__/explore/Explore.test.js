import { unmountComponentAtNode } from "react-dom";
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import "../../../__mocks__/server/matchMedia.mock";

jest.mock("react-responsive", () => ({
  useMediaQuery: jest.fn(),
}));

import { useInfiniteQuery, QueryClient } from "react-query";
jest.mock("react-query");

import Explore from "../../components/main/explore/Explore";

import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
const mockStore = configureStore([]);

import { _users } from "../../../__mocks__/server/users";
import { _user } from "../../../__mocks__/server/user";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("rendering Explore component", () => {
  it("renders Explore component", () => {
    const QueryClient = jest.fn();
    useInfiniteQuery.mockReturnValue({
      data: {},
      error: "",
      fetchNextPage: jest.fn(),
      hasNextPage: jest.fn(),
      isFetching: false,
      isFetchingNextPage: false,
      status: "loading",
    });

    const history = createMemoryHistory();

    let store = mockStore({
      dataReducer: {
        users: {},
      },
      authReducer: {
        user: _user,
      },
    });

    store.dispatch = jest.fn();

    const component = render(
      <Provider store={store}>
        <Router history={history}>
          <Explore />
        </Router>
      </Provider>
    );

    // screen.debug();
    expect(component.container).toMatchSnapshot();
    expect(screen.getByTestId("explore")).toBeInTheDocument();
  });
});
