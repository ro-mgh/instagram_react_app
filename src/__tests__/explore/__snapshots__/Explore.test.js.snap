// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Explore component renders Explore component 1`] = `
<div>
  <div
    data-testid="explore"
  >
    <div
      class="header-main"
      data-testid="header"
    >
      <div
        class="header-main-container"
      >
        <div
          class="header-logo-placeholder"
        >
          <a
            href="/"
          >
            <div
              class="header-logo-a"
            >
              <div
                class="customlogo-wrapper"
              >
                <div
                  class="header-logo-div"
                >
                  <img
                    alt=""
                    src="instagram_logo.png"
                  />
                </div>
                <div
                  class="customlogo-div"
                >
                   | by RM
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="header-search-placeholder"
        >
          <input
            autocapitalize="none"
            class="header-search-input"
            placeholder="Search"
            type="text"
            value=""
          />
          <div
            class="makeStyles-root-1"
          >
            <div />
          </div>
        </div>
        <div
          class="header-nav-placeholder"
        >
          <div
            class="header-nav-div"
          >
            <div
              class="header-nav-icon"
            >
              <a
                href="/"
              >
                <svg
                  aria-label="Home"
                  class=""
                  fill="#262626"
                  height="22"
                  viewBox="0 0 48 48"
                  width="22"
                >
                  <path
                    d="M45.5 48H30.1c-.8 0-1.5-.7-1.5-1.5V34.2c0-2.6-2.1-4.6-4.6-4.6s-4.6 2.1-4.6 4.6v12.3c0 .8-.7 1.5-1.5 1.5H2.5c-.8 0-1.5-.7-1.5-1.5V23c0-.4.2-.8.4-1.1L22.9.4c.6-.6 1.6-.6 2.1 0l21.5 21.5c.3.3.4.7.4 1.1v23.5c.1.8-.6 1.5-1.4 1.5z"
                  />
                </svg>
              </a>
            </div>
            <div
              class="header-nav-icon"
            >
              <a
                href="/explore"
              >
                <svg
                  aria-label="Find People"
                  class=""
                  fill="#262626"
                  height="22"
                  viewBox="0 0 48 48"
                  width="22"
                >
                  <path
                    clip-rule="evenodd"
                    d="M24 0C10.8 0 0 10.8 0 24s10.8 24 24 24 24-10.8 24-24S37.2 0 24 0zm0 45C12.4 45 3 35.6 3 24S12.4 3 24 3s21 9.4 21 21-9.4 21-21 21zm10.2-33.2l-14.8 7c-.3.1-.6.4-.7.7l-7 14.8c-.3.6-.2 1.3.3 1.7.3.3.7.4 1.1.4.2 0 .4 0 .6-.1l14.8-7c.3-.1.6-.4.7-.7l7-14.8c.3-.6.2-1.3-.3-1.7-.4-.5-1.1-.6-1.7-.3zm-7.4 15l-5.5-5.5 10.5-5-5 10.5z"
                    fill-rule="evenodd"
                  />
                </svg>
              </a>
            </div>
            <div
              class="header-nav-icon"
            >
              <div
                class="makeStyles-root-8"
              >
                <div>
                  <button
                    aria-haspopup="true"
                    class="MuiButtonBase-root MuiButton-root MuiButton-text"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <div
                        class="MuiAvatar-root MuiAvatar-circle makeStyles-small-10 MuiAvatar-colorDefault"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiAvatar-fallback"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="explore-wrapper"
    >
      <div
        class="explore-users-wrapper"
      >
        <p
          class="explore-users-text"
        >
          Discover People
        </p>
        <div
          class="explore-carousel-wrapper"
        >
          <div
            class=""
          >
            <div
              class="mainfield-progress-wrapper"
            >
              <div
                class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                role="progressbar"
                style="width: 30px; height: 30px;"
              >
                <svg
                  class="MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="explore-posts-wrapper"
      >
        <p
          class="explore-users-text"
        >
          Explore
        </p>
        <div
          class="mainfield-progress-wrapper"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 30px; height: 30px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <footer>
      <div
        class="footer-div-main"
      >
        <div
          class="footer-div-top"
        >
          <div
            class="footer-div-top-placeholder"
          >
            <a
              class="footer-text"
              href="https://about.instagram.com/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              About
            </a>
            <a
              class="footer-text"
              href="https://about.instagram.com/blog/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              Blog
            </a>
            <a
              class="footer-text"
              href="/about/jobs/"
            >
              Jobs
            </a>
            <a
              class="footer-text"
              href="https://help.instagram.com/"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              Help
            </a>
            <a
              class="footer-text"
              href="/developer/"
            >
              API
            </a>
            <a
              class="footer-text"
              href="/legal/privacy/"
            >
              Privacy
            </a>
            <a
              class="footer-text"
              href="/legal/terms/"
            >
              Terms
            </a>
            <a
              class="footer-text"
              href="/directory/profiles/"
            >
              Top Accounts
            </a>
            <a
              class="footer-text"
              href="/directory/hashtags/"
            >
              Hashtags
            </a>
            <a
              class="footer-text"
              href="/explore/locations/"
            >
              Locations
            </a>
          </div>
        </div>
        <div
          class="footer-div-bottom"
        >
          <div
            class="footer-text"
          >
            English
          </div>
          <div
            class="footer-text"
          >
            Â© 2021 Instagram by RM
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
